src/main/java/org/medicmobile/webapp/mobile/FilePickerHandler.java

```diff
@@ -0,0 +1,163 @@
+			sendDataToWebapp(null);
+			return;
+		}
+
+		sendDataToWebapp(intent);
+	}
+
+//> PRIVATE
+
+	/**
+	 * Executes the file picker callback.
+	 * The callback will give control back to WebView that automatically fetches the file by using a
+	 * provided URI, finally it assigns the bytes to the form's input. Even if there's not URI (ie null),
+	 * it needs to give control back to WebView, otherwise when clicking again on the form's input
+	 * it won't open the file picker.
+	 *
+	 * @param intent Intent from activity's response.
+	 */
+	private void sendDataToWebapp(Intent intent) {
+		if (this.filePickerCallback == null) {
+			warn(this, "FilePickerHandler :: Cannot send data back to webapp, filePickerCallback is null.");
+			return;
+		}
+
+		Uri uri = getSelectedFileUri(intent);
+		trace(this, "FilePickerHandler :: Sending data back to webapp, URI: %s", uri);
+		this.filePickerCallback.onReceiveValue(uri == null ? null : new Uri[] { uri });
+		this.filePickerCallback = null;
+	}
+
+	private Uri getSelectedFileUri(Intent intent) {
+		Uri uri = intent == null ? null : intent.getData();
+
+		if (uri == null && this.tempFileUri != null) {
+			uri = this.tempFileUri;
+			setTempFileUri(null);
+		}
+
+		return uri;
+	}
+
+	private void startFileCaptureActivity(FileChooserParams fileChooserParams) {
+		String[] mimeTypes = fileChooserParams.getAcceptTypes();
+		if (mimeTypes == null) {
+			warn(this, "FilePickerHandler :: MIME type is null, please specify a MIME type.");
+			sendDataToWebapp(null);
+			return;
+		}
+
+		trace(this, "FilePickerHandler :: Accepted MIME types: %s", Arrays.toString(mimeTypes));
+
+		Intent pickerIntent = new Intent(Intent.ACTION_GET_CONTENT);
+		pickerIntent.addCategory(Intent.CATEGORY_OPENABLE);
+		pickerIntent.putExtra(Intent.EXTRA_LOCAL_ONLY, true);
+		Intent captureIntent = null;
+
+		if (mimeTypes.length == 1) {
+			String mimeType = mimeTypes[0];
+			pickerIntent.setType(mimeType);
+			captureIntent = getCaptureIntent(mimeType);
+		} else {
+			pickerIntent.setType("*/*");
+			pickerIntent.putExtra(Intent.EXTRA_MIME_TYPES, mimeTypes);
+		}
+
+		Intent chooserIntent = Intent.createChooser(pickerIntent, this.context.getString(R.string.promptChooseFile));
+		// Ensure that there's an activity to handle the intent for capturing media.
+		if (captureIntent != null && captureIntent.resolveActivity(this.context.getPackageManager()) != null) {
+			chooserIntent.putExtra(Intent.EXTRA_INITIAL_INTENTS, new Intent[]{ captureIntent });
+		}
+
+		trace(this, "FilePickerHandler :: Starting activity with intents: chooser=%s, picker=%s, capture=%s",
+			chooserIntent, pickerIntent, captureIntent);
+
+		this.context.startActivityForResult(chooserIntent, RequestCode.FILE_PICKER_ACTIVITY.getCode());
+	}
+
+	private Intent getCaptureIntent(String mimeType) {
+		if (mimeType.contains("video/")) {
+			return new Intent(ACTION_VIDEO_CAPTURE);
+		}
+
+		if (mimeType.contains("audio/")) {
+			return new Intent(RECORD_SOUND_ACTION);
```

Comment:
This one is tricker to test, in my experience not all recording apps are set to open with this intent. 
I found that one called: "Audio Recorder" by "gitlab.com/axet" works with this.

Reply:
I got this app from playstore.